msg/OpenDroneIdArmStatus.msg

```diff
@@ -0,0 +1,3 @@
+uint64 timestamp
+uint8 status
```

Comment:
Shouldn't this be used in the preflight checks `openDroneIDCheck.cpp`, to check whether it equals to `MAV_ODID_ARM_STATUS_GOOD_TO_ARM`? Otherwise, the system would be able to arm without arm authorization.

Here's Ardupilot implementation for reference: https://github.com/ArduPilot/ardupilot/blob/5fa8b887a2df8a85822f84f41eec64bef5066895/libraries/AP_OpenDroneID/AP_OpenDroneID.cpp#L196C1-L198

Getting status message from GCS:
https://github.com/ArduPilot/ardupilot/blob/5fa8b887a2df8a85822f84f41eec64bef5066895/libraries/AP_OpenDroneID/AP_OpenDroneID.cpp#L748-L753

Note however, that I am a bit confused about what this message is actually supposed to do, as it also seems like QGC receives it (from the drone I think?). If so, is this simply used to transmit information whether system can be armed, from the drone?

Reply:
This is [OPEN_DRONE_ID_ARM_STATUS](https://mavlink.io/en/messages/common.html#OPEN_DRONE_ID_ARM_STATUS), the arm status of the remote id unit. 

Yes this should be used as a precondition for arming the vehicle, if remote ID is being used/required.

A GCS does not require this, but might reasonably use it to debug arming status problems. For example, the vehicle won't arm and you get an ODID error. However you don't know on the GCS what error actually is, and the message can tell you that in the error string. 
Now you might propagate this to the GCS using an event, but there is no such "standard" event, so you might say that the GCS publishing the error message is "more standard".